Fix graphics::Resource using VGC_DEBUG instead of VGC_DEBUG_BUILD (#1201)

dalboris opened the pull request:
#1201

`VGC_DEBUG` is for logging.

`VGC_DEBUG_BUILD` is for testing if the current build is a "Debug" build.

dalboris commented:
Note: 1st attempt of Linux build failed because Gitlab seemed temporarily down:

```
Fetching submodules
  /usr/bin/git submodule sync --recursive
  /usr/bin/git -c protocol.version=2 submodule update --init --force --depth=100 --recursive
  Submodule 'third/eigen' (https://gitlab.com/libeigen/eigen.git) registered for path 'third/eigen'
  Submodule 'third/fmt' (https://github.com/fmtlib/fmt.git) registered for path 'third/fmt'
  Submodule 'third/googletest' (https://github.com/google/googletest.git) registered for path 'third/googletest'
  Submodule 'third/libtess2' (https://github.com/vgc/libtess2.git) registered for path 'third/libtess2'
  Submodule 'third/pybind11' (https://github.com/pybind/pybind11.git) registered for path 'third/pybind11'
  Cloning into '/__w/vgc/vgc/third/eigen'...
  Cloning into '/__w/vgc/vgc/third/fmt'...
  Cloning into '/__w/vgc/vgc/third/googletest'...
  Cloning into '/__w/vgc/vgc/third/libtess2'...
  Cloning into '/__w/vgc/vgc/third/pybind11'...
  Error: fatal: unable to access 'https://gitlab.com/libeigen/eigen.git/': gnutls_handshake() failed: The TLS connection was non-properly terminated.
  Error: fatal: Fetched in submodule path 'third/eigen', but it did not contain 3147391d946bb4b6c68edd901f2add6ac1f31f8c. Direct fetching of that commit failed.
  Error: The process '/usr/bin/git' failed with exit code 128
``` 

Rerunning the job fixed it.

PixelRick commented:
Good find. I probably got mistaken trying to translate _DEBUG to its equivalent in vgc.
Maybe it is a good argument in favor of using `#if MACRO` instead of `#ifdef MACRO` and `#if defined(MACRO)`.
In C++ and C99, undefined identifiers in macros are converted to '0'.

dalboris commented:
@PixelRick I'm not even sure changing the convention to always use `#if` instead of `#ifdef` would prevent this mistake. In this case unintentionally writing `#if VGC_DEBUG` instead of `#if VGC_DEBUG_BUILD` would still compile... it just always evaluate to false instead of always evaluate to true when using `#ifdef`. The former makes the error more noticeable if we are developing in Debug mode, while the latter makes it more noticeable if we are developing in Release mode.

PixelRick commented:
Right, in a lot of cases it is a function-like macro that will evaluate to false and still compile. Unfortunate that is.
As you say, here it shifts the visibility of the problem between release and debug.
But more generally when the condition is incorrect in this way, is it better to always or never compile the block in ?

DEBUG alone is often used for builds when it should have been IS_DEBUG, DEBUGGING, DEBUG_BUILD, or BUILD_DEBUG.
I cannot tell if someone could do the same mistake in the future. But I don't think the odds are high enough to justify a change in the whole codebase.

Sorry for the mistake!

dalboris commented:
No worry :) Indeed, the mistake might happen again, but I don't think it's worth changing the code. Especially since the consequences of such mistake are so minor (just make the release build a tiny bit slower, most likely).

And you know what, it's as much my mistake as yours ^^

Not only because I reviewed it, but most importantly because I renamed VGC_DEBUG to VGC_DEBUG_BUILD on June 13, 2022:

https://github.com/vgc/vgc/commit/ee18775e9375ef997be578bcc59ed4cc4294c320

You committed the creation of the file resource.h using VGC_DEBUG on August 10, 2022:

https://github.com/vgc/vgc/commit/5c0fa5a3e23701eb3f030f9687cfa9ac81d88d85

So you probably just remembered the previous name, not realizing that I changed it. The reason I changed it is that it seems nice to have VGC_DEBUG, VGC_INFO, VGC_WARNING, etc., by analogy with Qt's qDebug(), qInfo(), qWarning(), etc. More explicit and less error-prone would be VGC_LOG_DEBUG, VGC_LOG_INFO, VGC_LOG_WARNING, but I thought it was a bit too verbose which could discourage their use.
